File: lib/mosek/11.0/tools/examples/java/helloworld.java
//
//  Copyright: Copyright (c) MOSEK ApS, Denmark. All rights reserved.
//
//  File:      helloworld.java
//
//  The most basic example of how to get started with MOSEK.


package com.mosek.example;
import mosek.*;

public class helloworld {
  public static void main(String[] args) {

    try (Task task = new Task()) {           // Create task

      task.appendvars(1);                          // 1 variable x
      task.putcj(0, 1.0);                          // c_0 = 1.0
      task.putvarbound(0, boundkey.ra, 2.0, 3.0);  // 2.0 <= x <= 3.0
      task.putobjsense(objsense.minimize);         // minimize

      task.optimize();                      // Optimize

      double x[] = task.getxx(soltype.itr);         // Get solution
      System.out.println("Solution x = " + x[0]);   // Print solution
    }
  }
}

File: lib/mosek/11.0/tools/examples/matlab/helloworld.m
%%
%  Copyright: Copyright (c) MOSEK ApS, Denmark. All rights reserved.
%
%  File:      helloworld.m
%
%  The most basic example of how to get started with MOSEK.
%

function [xx,prosta,solsta] = helloworld()
    model = mosekmodel(...
                  name = "helloworld", ...
                  objsense = "min", ...
                  objective = [ 1, 2 ]', ...
                  numvar = 2, ...
                  F = [ 1 0 ; ...
                        0 1 ; ...
                        3 -1 ], ...
                  domain = [ mosekdomain("greater than", rhs = 2), ...   % x >= 2,    1st row of F
                             mosekdomain("greater than", rhs = 3), ...   % y >= 3,    2nd row of F
                             mosekdomain("less than", rhs = 1) ]);       % 3x-y <= 1, 3rd row of F
    model.solve();
    
    if model.hassolution("interior")
        [xx,prosta,solsta] = model.getsolution("any", "x");
        xx
    end

end
